src/storage/storage.ts

```diff
@@ -134,6 +151,20 @@ export class Storage implements IStorage {
     return this.store.getAddressAtIndex(index);
   }
 
+  async getAddressPubkey(index: number): Promise<string> {
+    const addressInfo = await this.store.getAddressAtIndex(index);
+    if (addressInfo?.publicKey) {
+      // public key already cached on address info
+      return addressInfo.publicKey;
+    }
+
+    // derive public key from xpub
+    const accessData = await this._getValidAccessData();
+    const hdpubkey = new HDPublicKey(accessData.xpubkey);
+    const key: HDPublicKey = hdpubkey.deriveChild(index);
+    return key.publicKey.toString('hex');
```

Comment:
This is meant to be used to get the pubkey of the path `m/44'/280'/0'/0/index` where index is the address index.

When we sign the transaction with an external method we still need the actual pubkey to build the input data (on both multisig and p2pkh) but we cannot use the main key to derive it (which is the method we use on the lib).

Also, If the public key is generated during the initial address derivation we will use it (on the `addressInfo.publicKey`).

Reply:
Is there any downside to reusing the `deriveAddressFromXPubP2PKH()` function here instead of deriving manually?

https://github.com/HathorNetwork/hathor-wallet-lib/blob/91118d5335c2b63afb559e5deddc7a81f73cbc13/src/utils/address.ts#L27-L31 ?

Reply:
This is meant to be used by both p2pkh and p2sh wallets.
If you really want to minimize repetition the best approach would be to create an util `getAddressPubkeyFromXpub` and use it on both the `deriveAddressFromXPubP2PKH` and this new method.
But I think this is creates a new method for a very simple derive operation.